eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/bag/ImmutableBagFactory.java

```diff
@@ -49,6 +51,18 @@
 
     <T> ImmutableBag<T> with(T... elements);
 
+    <T> ImmutableBag<T> ofWithOccurrences(T element, int occurrence);
+
+    <T> ImmutableBag<T> ofWithOccurrences(T element1, int occurrence1, T element2, int occurrence2);
```

Comment:
I think we should have `withOccurrences` and `ofOccurrences` since we have `of` and `with` those 2 APIs. In this situation, the users will be much easier to find the new APIs.